raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -221,6 +224,20 @@ private void resetConnections() {
     private void onBecomeLeader(LeaderState state) {
         stateMachine.becomeLeader(quorum.epoch());
         updateLeaderEndOffset(state);
+
+        // Add a control message for faster high watermark advance.
+        if (!log.endOffsetForEpoch(quorum.epoch()).isPresent()) {
+            appendControlRecord(MemoryRecords.withLeaderChangeMessage(
+                time.milliseconds(),
+                quorum.epoch(),
+                new LeaderChangeMessageData()
+                    .setLeaderId(state.election().leaderId())
+                    .setGrantedVoters(
```

Comment:
My original thought in adding this field was to only add the voters that we received votes from. I thought this would be useful for debugging. Now that I'm thinking about it though, that might require us to store the granting voters in the quorum state file as well. Perhaps we can just remove this field?

Reply:
I'm neutral about this, but perhaps this does no harm at the moment for getting a redundant copy of the voted members when the quorum-state file is corrupted (which I agree is rare).

Reply:
I don't think it adds much value. However, if we want to keep it, I suggest we change the name to `Voters` instead of `GrantedVoters`.